### PR TITLE
Added fence.i after each memory load

### PR DESCRIPTION
**Pull request by p12nGH-zz**

### CODE REVIEW

**Comment by timsifive** [comment]
Are there case where we want to execute just a fence and not a fence.i? I'd be inclined to just add the fence_i to execute_fence(), and rename it to execute_fences().

Did you run `make all` in riscv-tests/debug to test for any regressions?

**Comment by p12nGH-zz** [comment]
It is a good idea to have execute_fences that does both types of fences.

**Review by mwachs5 — DISMISSED** [review]

File: src/target/riscv/program.c [hunk 1]
```diff
@@ -121,14 +121,11 @@ int riscv_program_csrw(struct riscv_program *p, enum gdb_regno s, enum gdb_regno
...
-int riscv_program_fence_i(struct riscv_program *p)
+int riscv_program_fences(struct riscv_program *p)
 {
-	return riscv_program_insert(p, fence_i());
-}
-
-int riscv_program_fence(struct riscv_program *p)
-{
-	return riscv_program_insert(p, fence());
+	if (riscv_program_insert(p, fence()) != ERROR_OK) return ERROR_FAIL;
```
Comment (timsifive) [inline src/target/riscv/program.c]: I believe the OpenOCD style guide does not allow for putting `return` on the same line is `if`. You can check with tools/checkpatch.sh (although that will only check the last git commit, and exit early until you add a Signed-off line).

**Review by timsifive — DISMISSED** [review]
I prefer to not modify riscv_program_fence_i(), but instead call
riscv_program_fence() and riscv_program_fence_i() from execute_fences(). This will allow us in the future to add some logic to execute_fences() to deal with a target that only has 1 word of program buffer, by executing first one and then the other.

File: src/target/riscv/riscv-013.c [hunk 2]
```diff
@@ -1436,6 +1436,30 @@ static int execute_fence(struct target *target)
 	return result;
 }
 
+static int execute_fence_i(struct target *target)
```
Comment (timsifive) [inline src/target/riscv/riscv-013.c:1439]: Do you still need this?

I think you can just replace all calls to execute_fence() with execute_fences().

**Review by timsifive — APPROVED** [review]

**Review by palmer-dabbelt — APPROVED** [review]
I think leaving `execute_fence_i()` doesn't hurt.

**Pull request closed without merging**